s3stream/src/main/java/com/automq/stream/s3/operator/AbstractObjectStorage.java

```diff
@@ -89,6 +89,19 @@ public AbstractObjectStorage(
         S3StreamMetricsManager.registerInflightS3WriteQuotaSupplier(inflightWriteLimiter::availablePermits, currentIndex);
     }
 
+    // used for test only
+    public AbstractObjectStorage(boolean manualMergeRead) {
```

Comment:
Prefer use a single base constructor and create a test constructor based on it. If there are many arguments in constructor, we could also use builder mode